microsoft-edge/webview2/how-to/javascript.md

```diff
@@ -30,6 +30,24 @@ Use the following functions to begin embedding JavaScript in your WebView app.
 | [ExecuteScriptAsync](/dotnet/api/microsoft.web.webview2.wpf.webview2.executescriptasync) | Run JavaScript in a WebView control. For more information, navigate to the Get Started tutorial. |
 | [OnDocumentCreatedAsync](/microsoft-edge/webview2/reference/win32/icorewebview2#addscripttoexecuteondocumentcreated) | Runs when the Document Object Model (DOM) is created. |
 
+<!-- ====================================================================== -->
+## Scenario: ExecuteScript JSON Encoded Results
+
+Because the result of `ExecuteScriptAsync` is JSON encoded, if the result evaluating the JavaScript is a string you will receive a JSON encoded string and not the value of the string. For example, the following script results in a string with the following value including the quotes at the start and end, and the escaping slashes. 
```

Comment:
```suggestion
Because the result of `ExecuteScriptAsync` is JSON encoded, if the result is a string you will receive a JSON encoded string and not the value of the string. For example, the following script results in a string with the following value including the quotes at the start and end, and the escaping slashes. 
```

Reply:
@david-risney - not sure what 'evaluating the JavaScript' means? do we need this. Also when we should define 'result', does this mean result as in return value of ExecuteScriptAsync or the value that is passed to script after it is json encoded?

Reply:
Right, result can sort of mean two things here I think: the result of running some JavaScript - a JavaScript object or the async string result of ExecuteScript/Async which is the JavaScript result that is then JSON encoded. 

For example:
* C# code: ExecuteScriptAsync("\"abc\"")
* Result (of script execution): "abc"
* Result (of ExecuteScriptAsync): "\"abc\""

The result of ExecuteScriptAsync is defined in terms of the result of script execution so we need to talk about both. I was trying to say the `result of evaluating the JavaScript` to distinguish it from the result of ExecuteScriptAsync but other wording or something would be good to.

Reply:
Changed the first sentence to "Because the result of `ExecuteScriptAsync` is JSON-encoded, if the result of evaluating the JavaScript is a string, you will receive a JSON-encoded string and not the value of the string." The ret of the paragraph matched Jason's edit.